### ISSUE TITLE
Proposal for Capybara/RSpec department

### ISSUE THREAD

**Issue by ydah** [rubocop/rubocop-capybara#47]
The `Capybara/SpecificMatcher`, `Capybara/NegationMatcher`, and `Capybara/VisibilityMatcher` cops depend on methods from the `Capybara::RSpecMatchers` class.

https://www.rubydoc.info/gems/capybara/Capybara/RSpecMatchers

While these cops were previously implemented in the RuboCop RSpec gem and did not cause any issues, they may not be appropriate for RuboCop Capybara, which is not limited to RSpec extensions.
Therefore, it might be a good idea to create a separate `Capybara/RSpec` department to handle these RSpec-specific extensions.

**Comment by ydah** [comment]
It might be a good idea to implement cops related to `[Capybara::Minitest::Assertions](https://www.rubydoc.info/gems/capybara/Capybara/Minitest/Assertions)` and `[Capybara::Minitest::Expectation](https://www.rubydoc.info/gems/capybara/Capybara/Minitest/Expectations)` as `Capybara/Minitest` department cops in a similar way to the proposed `Capybara/RSpec` department.

**Comment by G-Rath** [comment]
@ydah just wondering what the situation is with this? for reasons too silly to get into I have an app with dual rspec and minitest suites (as in equivalent tests in both suites), and so I realized the other day that the rspec version of a system test was triggering `Capybara/NegationMatcherAfterVisit` whereas the minitest version wasn't, and that lead me here...

When you say
> ...depend on methods from the `Capybara::RSpecMatchers` class.

is that still the case? I've had a look and nothing jumped out as using helpers directly related to RSpec, though obviously the cops themselves do have this assumption.

The bottom line is I can probably spare some bandwidth this/next month on trying to add some minitest support if you and the team have the bandwidth to review and potentially help with some of the surrounding setup for introducing `capybara/minitest`

(currently without having looked too deeply, my starting approach would be to just duplicate things into i.e. `capybarab/minitest/negation_matcher.rb` and explore any deduplicating later)

**Comment by ydah** [comment]
Basically, I believe the current `rubocop-capybara` assumes RSpec. This is because the original Cops for Capybara existed in `rubocop-rspec`. We welcome the addition of a Minitest department/division. For example, the relevant Cops will be placed under a method of a class such as the one shown next.
- https://www.rubydoc.info/gems/capybara/Capybara/Minitest/Assertions
- https://www.rubydoc.info/gems/capybara/Capybara/Minitest/Expectations
